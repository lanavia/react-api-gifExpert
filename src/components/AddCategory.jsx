import { useState } from "react"
import PropTypes from 'prop-types'


export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('')
    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()
        console.log(inputValue)
        if (inputValue.trim().length <= 1) return

        setInputValue('')
        onNewCategory(inputValue.trim())
    }
    return (
        /*  Remove (event)
        <form onSubmit={(event) => onSubmit(event)}>
             <input type="text"
                 placeholder="Buscar gifs"
                 value={inputValue}
                 onChange={(event) => onInputChange(event)} />
         </form> */

        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}