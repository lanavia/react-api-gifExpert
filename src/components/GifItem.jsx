import React from 'react'
import PropTypes from 'prop-types'
export const GifItem = ({ id, title, url }) => {
    return (
        <div className='card'>
            <p key={id}>{title}</p>
            <img src={url} />
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}