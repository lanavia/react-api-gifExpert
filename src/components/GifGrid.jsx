import { useEffect, useState } from 'react'
import { getGifs } from '../helpers/GetGifs'
import { GifItem } from './GifItem'


export const GifGrid = ({ category }) => {
    const [images, setImages] = useState([])

    const getImages = async () => {
        const newImages = await getGifs(category)
        setImages(newImages)
    }

    useEffect(() => {
        getImages()
    }, [])



    getGifs(category)
    return (
        <>
            <h3>{category}</h3>
            <div className='card-grid'>

                {/* opcion 1
                images.map(({ id, title, url }) => (

                    <GifItem key={id} title={title} url={url} />
                ))

                */}

                { //opcion 2 esparcir todas las propiedades de la imagen con el operador spread ...
                    images.map((image) => (

                        <GifItem key={image.id} {...image} />
                    ))

                }
            </div>
        </>
    )
}
