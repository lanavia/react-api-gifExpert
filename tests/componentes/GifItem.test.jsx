import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem/>', () => {
    const title = 'Saitama'
    const url = 'https://onepunchman.fandom.com/es/wiki/Saitama'
    test('Should match with snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })
}) 