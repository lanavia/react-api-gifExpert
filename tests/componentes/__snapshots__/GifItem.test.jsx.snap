// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Saitama
    </p>
    <img
      src="https://onepunchman.fandom.com/es/wiki/Saitama"
    />
  </div>
</div>
`;
